api/apps/api/src/modules/access-control/projects-acl/project-acl.service.ts

```diff
@@ -92,4 +94,109 @@ export class ProjectAclService implements ProjectAccessControl {
       this.canViewSolutionRoles,
     );
   }
+
+  async checkUserIsOwner(userId: string, projectId: string): Promise<Permit> {
+    return !!(await this.roles.findOne({
+      where: {
+        projectId,
+        userId,
+        roleName: Roles.project_owner,
+      },
+    }));
+  }
+
+  async checkLastOwner(userId: string, projectId: string): Promise<Permit> {
+    const allOwnersInProject = await this.roles.find({
+      where: {
+        projectId,
+        roleName: Roles.project_owner,
+        userId: Not(userId),
+      },
+    });
+    return allOwnersInProject.length >= 1;
+  }
+  /**
+   * @debt This module should not involve user details and it should deal with it
+   * using a standalone module that will access the data just to read it. We
+   * have to get back to it once scenarios, organizations and solutions are included
+   * inside the access-module.
+   */
+  async findUsersInProject(
+    projectId: string,
+    userId: string,
+  ): Promise<UsersProjectsApiEntity[] | Permit> {
+    if (!(await this.checkUserIsOwner(userId, projectId))) {
+      return false;
+    }
+
+    const usersInProject = await this.roles
+      .createQueryBuilder('users_projects')
+      .leftJoinAndSelect('users_projects.user', 'userId')
+      .where({ projectId })
+      .select([
+        'users_projects.roleName',
+        'userId.displayName',
+        'userId.id',
+        'userId.avatarDataUrl',
+      ])
+      .getMany();
+
+    return usersInProject;
+  }
+
+  async updateUserInProject(
+    projectId: string,
+    updateUserInProjectDto: UserRoleInProjectDto,
+    loggedUserId: string,
+  ): Promise<void | Permit> {
+    const { userId, roleName } = updateUserInProjectDto;
+    if (!(await this.checkUserIsOwner(loggedUserId, projectId))) {
+      return false;
+    }
+
+    const apiDbConnection = getConnection(DbConnections.default);
+    const apiQueryRunner = apiDbConnection.createQueryRunner();
+
+    await apiQueryRunner.connect();
+    await apiQueryRunner.startTransaction();
```

Comment:
Consider using `EntityManager` (`manager.transaction( transaction => .... )` ) to get rid of boilerplate code, including manually getting the connection.